src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -277,14 +272,17 @@ def _retrieve_unit_table_pynwb(nwbfile: "NWBFile", unit_table_path: Optional[str
 
 
 def _is_hdf5_file(filename_or_file):
-    # Source for magic numbers https://www.loc.gov/preservation/digital/formats/fdd/fdd000229.shtml
-    # We should find a better one though
     if isinstance(filename_or_file, (str, Path)):
-        with open(filename_or_file, "rb") as f:
-            file_signature = f.read(8)
+        import h5py
+
+        filename = str(filename_or_file)
+        is_hdf5 = h5py.h5f.is_hdf5(filename.encode("utf-8"))
     else:
         file_signature = filename_or_file.read(8)
-    return file_signature == b"\x89HDF\r\n\x1a\n"
+        # Source of the magic number https://docs.hdfgroup.org/hdf5/develop/_f_m_t3.html
```

Comment:
@alejoe91 found a better source plus a native method to sort this out when the file_path is provided.